src/Services/Azure/Kusto/KustoClient.cs

```diff
@@ -0,0 +1,91 @@
+        var httpRequest = await GenerateHttpRequestMessage(uri, database, text, cancellationToken).ConfigureAwait(false);
+
+        using var httpClient = _httpClientFactory.CreateClient();
+        httpClient.BaseAddress = new Uri(_clusterUri);
+        return await SendRequestAsync(httpClient, httpRequest, cancellationToken).ConfigureAwait(false);        
+    }
+
+    public async Task<KustoResult> ExecuteControlCommandAsync(string database, string text, CancellationToken cancellationToken)
+    {
+        var uri = _clusterUri + "/v1/rest/mgmt";
+        var httpRequest = await GenerateHttpRequestMessage(uri, database, text, cancellationToken).ConfigureAwait(false);
+
+        var httpClient = _httpClientFactory.CreateClient();
+        httpClient.BaseAddress = new Uri(_clusterUri);
+        return await SendRequestAsync(httpClient, httpRequest, cancellationToken).ConfigureAwait(false);
+    }
+    #endregion
+
+    #region Private Methods
+    private async Task<HttpRequestMessage> GenerateHttpRequestMessage(string uri, string database, string text, CancellationToken cancellationToken = default)
+    {
+        var httpRequest = new HttpRequestMessage(HttpMethod.Post, uri);
+
+        // Auth info
+        var scopes = new string[]
+        {
+            s_default_scope
+        };
+        var clientRequestId = s_clientRequestIdPrefix + Guid.NewGuid().ToString();
+        var tokenRequestContext = new TokenRequestContext(scopes, clientRequestId);
+        var accessToken = await _tokenCredential.GetTokenAsync(tokenRequestContext, cancellationToken);
+        httpRequest.Headers.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("bearer", accessToken.Token);
+
+        httpRequest.Headers.Add("x-ms-client-request-id", clientRequestId);
+        httpRequest.Headers.Add("x-ms-app", s_application);
+        httpRequest.Headers.Add("x-ms-client-version", "Kusto.Client.Light"); // Kusto.Dotnet.Client:{13.0.2}|Runtime:{.NETv4.8.1/CLRv4.0.30319.42000/4.8.9300.0_built_by:_NET481REL1LAST_C}
+        httpRequest.Headers.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+
+        // Body
+        var body = new JsonObject();
+        body.Add("db", database);
+        body.Add("csl", text);
+        var properties = new JsonObject();
+        properties.Add("ClientRequestId", clientRequestId); // TODO: Also add this as a header?
+        body.Add("properties", properties);
+        var bodyStr = body.ToJsonString();
+        httpRequest.Content = new StringContent(bodyStr);
+        httpRequest.Content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/json", "utf-8");
+        return httpRequest;
+    }
+
+    private async Task<KustoResult> SendRequestAsync(HttpClient httpClient, HttpRequestMessage httpRequest, CancellationToken cancellationToken = default)
+    {
+        var httpCompletionOption = new HttpCompletionOption();
```

Comment:
nit; no need to create option for each request, we can directly use `HttpCompletionOption.ResponseContentRead`